bagel/tests/test_utility.py

```diff
@@ -543,7 +574,216 @@ def test_failed_json_reading_raises_informative_error(
 def test_unsupported_tsv_encoding_raises_informative_error(test_data, capsys):
     """Test that given an input phenotypic TSV with an unsupported encoding, the CLI exits with an informative error message."""
     with pytest.raises(typer.Exit):
-        putil.load_pheno(test_data / "example_iso88591.tsv")
+        futil.load_tabular(test_data / "example_iso88591.tsv")
     captured = capsys.readouterr()
 
     assert "Failed to decode the input file" in captured.err
+
+
+def test_extract_subs_from_jsonld_dataset(
+    test_data_upload_path, load_test_json
+):
+    """Test that subjects are correctly extracted from a JSONLD dataset."""
+    dataset = load_test_json(
+        test_data_upload_path / "example_synthetic.jsonld"
+    )
+    dataset.pop("@context")
+    subjects = extract_subs_from_jsonld_dataset(
```

Comment:
Why not just make a models.Dataset instance here with some known subjects and then feed that in? 
My concern is that we're now testing two functions (however simple), and you assert over "is it a subject instance"
instead of "do we have the right subject labels" which is arguable more interesting (but obscured by loading the external file).